Explorer/Assets/Scripts/ECS/SceneLifeCycle/ECSReloadScene.cs

```diff
@@ -55,15 +62,65 @@ public async UniTask<bool> TryReloadSceneAsync(CancellationToken ct)
         private async UniTask DisposeAndRestartAsync(Entity entity, ISceneFacade currentScene, CancellationToken ct)
         {
             ct.ThrowIfCancellationRequested();
+
             //There is a lingering promise we need to remove, and add the DeleteEntityIntention to make the standard unload flow.
             world!.Remove<AssetPromise<ISceneFacade, GetSceneFacadeIntention>>(entity);
             world.Add<DeleteEntityIntention>(entity);
 
             //We wait until scene is fully disposed
             await UniTask.WaitUntil(() => currentScene.SceneStateProvider.State.Equals(SceneState.Disposed), cancellationToken: ct);
 
-            //Forcing a fake dirtyness to force a reload of the scene
-            world.Get<PartitionComponent>(entity).IsDirty = true;
+            if (localSceneDevelopment)
+            {
+                await ReloadCurrentSceneFromScratchAsync(currentScene);
+            }
+            else
+            {
+                // Forcing a fake IsDirty to force a reload of the scene at ResolveVisualSceneStateSystem.AddSceneVisualStateQuery()
+                world.Get<PartitionComponent>(entity).IsDirty = true;
+            }
+        }
+
+        private async UniTask ReloadCurrentSceneFromScratchAsync(ISceneFacade currentScene)
+        {
+            var baseParcel = currentScene.Info.BaseParcel;
+            world.Query(in new QueryDescription().WithAll<RealmComponent>(),
+                (ref ProcessedScenePointers processedPointers,
+                    ref VolatileScenePointers volatileScenePointers,
+                    ref RealmComponent realmComponent) =>
+                {
+                    // Remove pointers so that LoadScenePointerSystemBase.TryCreateSceneEntity() processes the new scene instance correctly
+                    foreach (Vector2Int parcel in currentScene.SceneData.Parcels)
+                    {
+                        processedPointers.Value.Remove(parcel.ToInt2());
+                    }
+
+                    // Add promise for loading the scene from scratch again at LoadPointersByIncreasingRadiusSystem
+                    var inputList = new List<int2>();
+                    inputList.Add(baseParcel.ToInt2());
+                    volatileScenePointers.ActivePromise
```

Comment:
Shouldnt you have to cancel the promise in case it is already running? Also perhaps you can extract this part into a method, something like: `RestartScenePromise`

Reply:
I removed that part based on Juani's suggestion.